src/Site_Command.php

```diff
@@ -265,7 +265,12 @@ function delete( $args, $assoc_args ) {
 
 		$site_url = trailingslashit( $blog->siteurl );
 
-		WP_CLI::confirm( "Are you sure you want to delete the '$site_url' site?", $assoc_args );
+		if ( $blog_id == BLOG_ID_CURRENT_SITE && ! \WP_CLI\Utils\get_flag_value( $assoc_args, 'delete-root' ) ) {
+			WP_CLI::line( WP_CLI::colorize( "%R'$site_url' is the root site. Pass the --delete-root flag to delete it.%n" ), $assoc_args );
+			return;
+		} else {
+			WP_CLI::confirm( "Are you sure you want to delete the '$site_url' site?", $assoc_args );
+		}
```

Comment:
Given we already have a confirmation prompt before deleting a site, I don't think it's necessary to require another flag to delete the root site.

Instead, when deleting the root site we should customize the confirmation message to indicate what happens when you delete the root site.

As currently implemented, this would be a breaking change

Reply:
For a scenario as:

```
http://wpcli.com
http://one.wpcli.com
http://two.wpcli.com
```
Following observations have been made after executing
`wp site delete 1`

1. The tables aren't dropped but the users from `wpcli.com` blog are deleted.
2. `wpcli.com` gets deactivated ( You can only activate it again through WP-CLI since the admin dashboard doesn't give you an option to activate/deactivate the main site. )

Irrelevant of the fact that whether you want to keep/drop tables, the users will get deleted as evident [here](https://core.trac.wordpress.org/browser/tags/4.8/src/wp-admin/includes/ms.php#L82)